Core dump due to Invalid UTF-8 error

**Information**:
System: Arch Linux
Install method: AUR sioyek-git
Sioyek version: 2.0.0.r1041.g280fdb6-1
Libmupdf version: 1.26.1-1

**Error description**:
On attempting to open [this file](https://mega.nz/file/r8QVDDoA#iyy2f5MjwTFpQxBk6sqBwSc7bDGVjg510YgiRYRMO20), Sioyek core dumps due to an "Invalid UTF-8" error. I have not been able to find other files that produce this error; they open just fine. The problem file opens just fine using other PDF viewers.

**Error message**:
```
default_config_path: /etc/sioyek/prefs.config
default_keys_path: /etc/sioyek/keys.config
user_config_path: [ 0 ] /etc/xdg/sioyek/prefs_user.config
user_config_path: [ 1 ] /home/shadi/.config/sioyek/prefs_user.config
user_keys_path: [ 0 ] /etc/xdg/sioyek/keys_user.config
user_keys_path: [ 1 ] /home/shadi/.config/sioyek/keys_user.config
database_file_path: /home/shadi/.local/share/sioyek/test.db
local_database_file_path: /home/shadi/.local/share/sioyek/local.db
global_database_file_path: /home/shadi/.local/share/sioyek/shared.db
tutorial_path: /usr/share/sioyek/tutorial.pdf
last_opened_file_address_path: /home/shadi/.local/share/sioyek/last_document_path.txt
shader_path: /usr/share/sioyek/shaders
SIOYEK
Warning: key defined in /etc/sioyek/keys.config:288 overwritten by /home/shadi/.config/sioyek/keys_user.config:3. Overriding command: q: replacing quit with close_window
Warning: key defined in /etc/sioyek/keys.config:255 overwritten by /home/shadi/.config/sioyek/keys_user.config:5. Overriding command: s: replacing external_search with fit_to_page_width
Warning: key defined in /etc/sioyek/keys.config:227 overwritten by /home/shadi/.config/sioyek/keys_user.config:13. Overriding command: <tab>: replacing goto_portal with goto_toc
terminate called after throwing an instance of 'utf8::invalid_utf8'
  what():  Invalid UTF-8
zsh: IOT instruction (core dumped)  sioyek hobsbawm-eric-the-invention-of-tradition.pdf
```
**Steps taken**:
I could not find any similar issues on GitHub. I could not attempt a fix myself because I don't understand this error.

`./build_linux.sh` is giving me the following error: `make: *** No targets specified and no makefile found.  Stop.`

Sorry, I'm a bit of a beginner.

EDIT: per #730, passed `--recursive` and currently building with no issues
EDIT: Had to set `QMAKE="/usr/bin/qmake6"` right after the nested `if` in `build_linux.sh` to abjure the `Project ERROR: Unknown module(s) in QT: texttospeech`. Build successful. 

EDIT: For future reference, to build from the dev branch, the following commands should work:
```
git clone --recursive https://github.com/ahrm/sioyek.git -b development
cd sioyek
QMAKE=qmake6 ./build_linux.sh
```
The final build is in `./build/sioyek`.

--- a/pdf_viewer/document.cpp
+++ b/pdf_viewer/document.cpp
@@ -23,6 +23,7 @@
 #include "sqlite3.h"
 #include "checksum.h"
 #include "database.h"
+#include "utf8/checked.h"
 #include "utils.h"
 
 extern bool SHOULD_RENDER_PDF_ANNOTATIONS;
@@ -821,35 +822,39 @@ void Document::create_toc_tree(std::vector<TocNode*>& toc) {
 void Document::convert_toc_tree(fz_outline* root, std::vector<TocNode*>& output) {
     // convert an fz_outline structure to a tree of TocNodes
 
-    std::vector<int> accum_chapter_pages;
-    count_chapter_pages_accum(accum_chapter_pages);
-
-    do {
-        if (root == nullptr || root->title == nullptr) {
-            break;
-        }
-
-        TocNode* current_node = new TocNode;
-        current_node->title = utf8_decode(root->title);
-        current_node->x = root->x;
-        current_node->y = root->y;
-        if (root->page.page == -1) {
-            float xp, yp;
-            fz_location loc = fz_resolve_link(context, doc, root->uri, &xp, &yp);
-            int chapter_page = 0;
-            if (loc.chapter >= 0 && loc.chapter < accum_chapter_pages.size()) {
-                chapter_page = accum_chapter_pages[loc.chapter];
+    try {
+        std::vector<int> accum_chapter_pages;
+        count_chapter_pages_accum(accum_chapter_pages);
+        do {
+            if (root == nullptr || root->title == nullptr) {
+                break;
             }
-            current_node->page = chapter_page + loc.page;
-        }
-        else {
-            current_node->page = root->page.page;
-        }
-        convert_toc_tree(root->down, current_node->children);
+
+                TocNode* current_node = new TocNode;
+                current_node->title = utf8_decode(root->title);
+                current_node->x = root->x;
+                current_node->y = root->y;
+            if (root->page.page == -1) {
+                float xp, yp;
+                fz_location loc = fz_resolve_link(context, doc, root->uri, &xp, &yp);
+                int chapter_page = 0;
+                if (loc.chapter >= 0 && loc.chapter < accum_chapter_pages.size()) {
+                    chapter_page = accum_chapter_pages[loc.chapter];
+                }
+                current_node->page = chapter_page + loc.page;
+            }
+            else {
+                current_node->page = root->page.page;
+            }
+            convert_toc_tree(root->down, current_node->children);
 
 
-        output.push_back(current_node);
-    } while ((root = root->next));
+            output.push_back(current_node);
+        } while ((root = root->next));
+    }
+    catch(utf8::exception& e) {
+        std::cerr << "could not convert toc node: " << e.what() << std::endl;
+    }
 }
 
 PdfLink Document::merge_links(const std::vector<PdfLink>& links_to_merge) {

